pkg/timeseries/dataset/dataset.go

```diff
@@ -419,8 +419,8 @@ func (ds *DataSet) DefaultRangeCropper(e timeseries.Extent) {
 		if len(ds.Results[i].SeriesList) == 0 {
 			continue
 		}
-		sl := make([]*Series, 0, len(ds.Results[i].SeriesList))
-		for _, s := range ds.Results[i].SeriesList {
+		sl := make(chan seriesResult, len(ds.Results[i].SeriesList))
```

Comment:
is it possible to resolve this race without using a channel? given that channels are quite slow (and much slower than just using a mutex), we currently/intentionally do not use them anywhere in the HTTP request processing pipeline.

Reply:
we _could_ guard the slice updates with a mutex; let me try this out perhaps with a benchmark.

Reply:
Migrated to mutex.

Existing benchmark:

---
Main:
```
BenchmarkCropToRange-12    	55737582	      1621 ns/op
PASS
ok  	github.com/trickstercache/trickster/v2/pkg/timeseries/dataset	201.594s
```
---
Mutex:
```
BenchmarkCropToRange-12    	 6267260	       657.0 ns/op
PASS
ok  	github.com/trickstercache/trickster/v2/pkg/timeseries/dataset	18.047s
```
-----
channel:
```
BenchmarkCropToRange-12    	54511280	       996.8 ns/op
PASS
ok  	github.com/trickstercache/trickster/v2/pkg/timeseries/dataset	152.904s
```